src/Swarm/TUI/View.hs

```diff
@@ -326,9 +326,31 @@ drawGameUI s =
   moreTop = s ^. uiState . uiMoreInfoTop
   moreBot = s ^. uiState . uiMoreInfoBot
 
-drawWorldCursorInfo :: Bool -> GameState -> W.Coords -> Widget Name
-drawWorldCursorInfo showRobots g i@(W.Coords (y, x)) =
-  hBox [drawLoc showRobots g i, txt $ " at " <> from (show x) <> " " <> from (show (y * (-1)))]
+drawWorldCursorInfo :: GameState -> W.Coords -> Widget Name
+drawWorldCursorInfo g coords@(W.Coords (y, x)) =
+  hBox $ tileMemberWidgets ++ [coordsWidget]
+ where
+  coordsWidget =
+    txt $
+      T.unwords
+        [ from $ show x
+        , from $ show $ y * (-1)
+        ]
+
+  tileMembers = terrain : mapMaybe merge [entity, robot]
+  tileMemberWidgets =
+    map (padRight $ Pad 1) $
+      concat $
+        reverse $
+          zipWith f tileMembers ["at", "on", "with"]
```

Comment:
The prepositions are ordered backwards here so that `zipWith` will only pick off just `at`, or  `on` plus `at`, if we only have one or two members on the tile.